Fw/Types/StringType.cpp

```diff
@@ -49,7 +49,7 @@ bool StringBase::operator==(const CHAR* other) const {
     }
 
     const SizeType capacity = this->getCapacity();
-    const size_t result = strncmp(us, other, capacity);
+    const size_t result = static_cast<size_t>(strncmp(us, other, capacity));
```

Comment:
For future review: do we need to cast to `size_t` just for a comparison to `0` below?  Perhaps we should just use the native result type from `strncmp`